src/assets/htmx.md

```diff
@@ -0,0 +1,347 @@
+
+```html
+<div hx-post="/hello">hello</div>
+```
+
+[demo](https://i.imgur.com/BidNTh3.gif)
+
+As seen above the endpoint is working and the response is shown properly in the hello example.
+Here clicking the hello text will send a request that is because we did not define any event that will trigger the ajax request.
+
+We can explicitly define it like
+```html
+<div hx-get="/hello" hx-trigger="click">hello</div>
+```
+
+Htmx has other predefined events like
+
+- `hx-trigger="mouseenter"` triggers when the mouse enters the element
+- `hx-trigger="mouseenter once"` triggers when the mouse enters the element and won't trigger after that
+- `hx-trigger="click[ctrlKey]"` triggers when the element is clicked while holding the `Ctrl` key
+- `hx-trigger="every 2s"` triggers every two seconds
+
+You can find more on the htmx docs.
+
+Currently, it replaces the element that makes the ajax request similar to the trigger event we have a `hx-target` attribute
+That specifies which element is going to get replaced
+
+```html
+<div id="relpace" class="replace"></div>
+<div hx-get="/hello" hx-trigger="click" hx-target="#relpace">hello</div>
+```
+
+We can also use class like `.replace` or give elements like `body` as target
+
+For example:
+If we have two elements `foo`
+
+```html
+<foo id="1" class="replace"></foo>
+<foo id="2" class="replace"></foo>
+<bar hx-post="/hello" hx-trigger="click" hx-target="foo">hello</bar>
+```
+
+It swaps the first of the `foo` element with `id=1`
+
+
+## making a todo app
+Since we learned a bit of basic let's write a todo app.
+
+First start with making a submit form for adding a todo task
+
+For that, we will only need an input box and a button
+
+```html
+<div id="app" class="container">
+  <div id="todo-list"></div>
+  <form
+    id="myForm"
+    hx-post="/save"
+    hx-trigger="submit"
+    hx-target="#todo-list"
+    hx-swap="beforeend"
+  >
+    <input type="text" name="task" required />
+    <input type="submit" value="save" />
+  </form>
+</div>
+```
+
+Now on the server
+
+```go
+func save(w http.ResponseWriter, r *http.Request) {
+    r.ParseForm()
+    storeTODO(r.FormValue("task")) // save to database
+    tastList := "<div>" + r.FormValue("task") + " </div>"
+    io.WriteString(w, tastList)
+}
+
+func main() {
+    ...
+    http.HandleFunc("/save", save)
+    ...
+}
+```
+
+[demo](https://i.imgur.com/wmTHPYD.gif)
+
+
+In the above html, the form will send a post request to `/save` endpoint and htmx will replace the content in the div with the id `todo-list` but the whole content will be replaced.
+
+```html
+<div id="todo-list">
+  <div>item 1</div>
+</div>
+```
+
+After adding another
+
+```html
+<div id="todo-list">
+  <div>item 2</div>
+</div>
+```
+
+Instead, we want to only append the new todo item after the last todo item. so we use the attribute `hx-swap` to let htmx know where we want the response to render. the `beforeend` value will append the html response before the end of the targeted element
```

Comment:
ether the `.` are end of sentence then the first word after that should be written with a capital letter or they should be `,`.